libs/SmartSync/src/com/salesforce/androidsdk/smartsync/target/SyncUpTarget.java

```diff
@@ -176,6 +199,11 @@ public String createOnServer(SyncManager syncManager, JSONObject record, List<St
     protected String createOnServer(SyncManager syncManager, String objectType, Map<String, Object> fields) throws IOException, JSONException {
         RestRequest request = RestRequest.getRequestForCreate(syncManager.apiVersion, objectType, fields);
         RestResponse response = syncManager.sendSyncWithSmartSyncUserAgent(request);
+
+        if (!response.isSuccess()) {
+            lastError = response.asString();
```

Comment:
I chose to save the value in an instance variable to avoid changing the signatures of a bunch of methods.